src/vs/workbench/browser/parts/editor/tabsTitleControl.ts

```diff
@@ -174,6 +174,9 @@ export class TabsTitleControl extends TitleControl {
 		this.tabsContainer.setAttribute('role', 'tablist');
 		this.tabsContainer.draggable = true;
 		this.tabsContainer.classList.add('tabs-container');
+		this.tabsContainer.addEventListener('mouseenter', this.fixTabWidths.bind(this));
+		this.tabsContainer.addEventListener('mouseleave', this.relaxTabWidths.bind(this));
+		this.updateTabSizingMaxWidth();
```

Comment:
Also this call is only needed when the option is on?

Reply:
I've updated the inside of the function so it's much clearer that it does things depending on the setting; it needs to have the condition inside because it's called also on settings update. Do you really think it's worth adding the two lines of code here?